forest/shared/src/cli/snapshot_fetch.rs

```diff
@@ -245,6 +325,13 @@ async fn snapshot_fetch_filecoin(
     create_dir_all(snapshot_out_dir).await?;
     let snapshot_name = normalize_filecoin_snapshot_name(&config.chain.name, &filename)?;
     let snapshot_path = snapshot_out_dir.join(&snapshot_name);
+    let total_size = snapshot_response
+        .headers()
+        .get("content-length")
+        .and_then(|ct_len| ct_len.to_str().ok())
+        .and_then(|ct_len| ct_len.parse::<u64>().ok())
+        .ok_or_else(|| anyhow::anyhow!("Couldn't retrieve content length"))?;
```

Comment:
open question: not having a `content-length` header is not a reason to bail out i guess? A server may not provide content length but can still can send bytes using streaming responses.

Reply:
True, but in this particular exercise, the content length is *required*, since we are not downloading the file (yet), but advertising the size.